tests/test_docker.py

```diff
@@ -181,17 +180,18 @@ def test_build_with_buildargs(self):
             version='auto',
             timeout=180)
 
-        # To capture all output to inspect, must delay removal until
-        # after retrieval of logs otherwise the API can sometimes return
-        # an empty result
-        c = client.containers.create(im)
         try:
+            # To capture all output to inspect, must delay removal until
+            # after retrieval of logs otherwise the API can sometimes return
+            # an empty result
+            c = client.containers.create(im)
```

Comment:
if this fails than this will cause unnecessary exceptions in finally clause, as "create" would never complete and variable c would not get defined/set to right value.

Reply:
Actually yes, I missed that we are also closing client. Maybe we should just make dockerclient to be a fixture?

Reply:
Just used a `addCleanup(client.close)` call for now, if there is more call for the docker client in that test class we can create a proper fixture in the future.